### ISSUE TITLE
iOS touch inputs not working after calling SKStoreReviewController.requestReview();

### ISSUE THREAD

**Issue by CoderBaron** [libgdx/libgdx#6769]
#### Issue details
When you call SKStoreReviewController.requestReview() or requestReviewInScene() and close the opening view controller, the app will not respond to touch inputs anymore. It doesn't matter if you call the function from the libGDX or iOS main thread. 
However rendering continues to work normally.

#### Reproduction steps/code
Create a new game with the libGDX setup jar. Modify the following generated classes.

MyGdxGame.java:
```
package com.mygdx.game;

import com.badlogic.gdx.ApplicationAdapter;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.InputProcessor;
import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.utils.ScreenUtils;

public class MyGdxGame extends ApplicationAdapter implements InputProcessor {
	SpriteBatch batch;
	Texture img;
	int lastScreenX, lastScreenY;
	
	@Override
	public void create () {
		batch = new SpriteBatch();
		img = new Texture("badlogic.jpg");
		Gdx.input.setInputProcessor(this);
	}

       @Override
	public void resume() {
		super.resume();
		lastScreenX = Gdx.graphics.getWidth() / 2;
		lastScreenY = Gdx.graphics.getHeight() / 2;
	}

	@Override
	public void render() {
		ScreenUtils.clear(Color.BLACK);
		batch.begin();
		batch.draw(img, lastScreenX, lastScreenY);
		batch.end();
	}

	@Override
	public boolean keyDown(int keycode) {
		return false;
	}

	@Override
	public boolean keyUp(int keycode) {
		return false;
	}

	@Override
	public boolean keyTyped(char character) {
		return false;
	}

	@Override
	public boolean touchDown(int screenX, int screenY, int pointer, int button) {
		lastScreenX = screenX;
		lastScreenY = Gdx.graphics.getHeight() - screenY;
		return false;
	}

	@Override
	public boolean touchUp(int screenX, int screenY, int pointer, int button) {
		return false;
	}

	@Override
	public boolean touchDragged(int screenX, int screenY, int pointer) {
		return false;
	}

	@Override
	public boolean mouseMoved(int screenX, int screenY) {
		return false;
	}

	@Override
	public boolean scrolled(float amountX, float amountY) {
		return false;
	}
}
```

IOSLauncher.java:
```
package com.mygdx.game;

import com.badlogic.gdx.backends.iosrobovm.IOSApplication;
import com.badlogic.gdx.backends.iosrobovm.IOSApplicationConfiguration;
import org.robovm.apple.foundation.NSAutoreleasePool;
import org.robovm.apple.storekit.SKStoreReviewController;
import org.robovm.apple.uikit.UIApplication;

public class IOSLauncher extends IOSApplication.Delegate {
    @Override
    protected IOSApplication createApplication() {
        IOSApplicationConfiguration config = new IOSApplicationConfiguration();
        return new IOSApplication(new MyGdxGame(), config);
    }

    public static void main(String[] argv) {
        NSAutoreleasePool pool = new NSAutoreleasePool();
        UIApplication.main(argv, null, IOSLauncher.class);
        pool.close();
    }

    boolean first;

    @Override
    public void didBecomeActive(UIApplication application) {
        super.didBecomeActive(application);

        if (!first) {
            first = true;
        } else {
            SKStoreReviewController.requestReviewInScene(getWindow().getWindowScene());
        }
    }
}
```

You have to run this test app on an iOS device optimally with iOS 15 (that's the version I used). Use a wildcard provisioning profile or edit the app.id in the robovm.properties file to one of your registered iOS app ids for quick testing. If you use one of your app ids your app will be deleted from your device when installing this test app.

When you run the app you can tap the screen to move the image around. Minimize the app, then reopen it and the review view controller should open. Dismiss it and voilà - touch input doesn't work anymore. You can tap around but the image doesn't move.
To find out where this issue originates I would put breakpoints into IOSGraphics.java line 111 and 126 (touchesBegan, touchesMoved). This should normally be called whenever you touch the screen. If this doesn't get called anymore after dismissing the reviewcontroller, it means that the issue is with the GLKView and how it is setup. If it does get called, then it's certainly an issue how libGDX handles iOS touch input. 

I would test it myself, unfortunately my debug setup for iOS doesn't work. I can't get the debugger to work.

#### Version of libGDX and/or relevant dependencies
1.10.0

#### Please select the affected platforms
- [ ] Android
- [X] iOS
- [ ] HTML/GWT
- [ ] Windows
- [ ] Linux
- [ ] macOS

For now I'm removing the review feature from my apps because I don't want my apps to freeze. I hope someone with working iOS debug setup can nail down this issue.

**Comment by Berstanio** [comment]
Is that a similiar issue to the one you mentioned on the MetalANGLE pr with the simulator?

**Comment by obigu** [comment]
Looks like it may not be libGDX specific
https://developer.apple.com/forums/thread/692923
https://stackoverflow.com/questions/69272004/closing-rate-and-review-on-ipados-15-freezes-interface

**Comment by CoderBaron** [comment]
> Is that a similiar issue to the one you mentioned on the MetalANGLE pr with the simulator?

This could be related. Seems iOS 15 is having more issues with freezing apps. I'll have to try with iOS 14 to rule out any issue with your Metal PR.

**Comment by CoderBaron** [comment]
Issue could be due to "App Prewarming" being introduced in iOS 15: https://developer.apple.com/documentation/uikit/app_and_environment/responding_to_the_launch_of_your_app/about_the_app_launch_sequence?language=objc#3894431

There may be some steps needed to make libGDX and/or RoboVM fully compatible and working with iOS 15.

**Comment by obigu** [comment]
I've been testing this and can reproduce.
- It occurs both on device and simulator.
- The "freeze" always occurs, no matter when you show the rating dialog (it's not due to being displayed too early on `didBecomeActive`)

As described by OP, after dismissing the dialog, touches are not received anymore by the app. The `touchesXXXX` methods are not even being called which means no touch events are being receiceived. I've also noticed `pressedXXXX` than trigger when a physical keyboard is pressed also stop being received which would suggest it is not an issue of some find of invisible overlay blocking touches from getting through. 

I've tried determining who is the `firstResponder` on a libGDX app see if it changed after displaying the dialog but haven't found it (the `View`, the `ViewController` and the `UIApplication` return `false`).

I've tried several things without success and I'm running out of ideas, may be worth asking RoboVM god dkmitsa.
